cli/tests/test_completion.rs

```diff
@@ -741,6 +741,59 @@ fn test_template_alias() {
     ");
 }
 
+#[test]
+fn test_merge_tools() {
+    let mut test_env = TestEnvironment::default();
+    test_env.add_env_var("COMPLETE", "fish");
+    let dir = test_env.env_root();
+
+    let output = test_env.run_jj_in(dir, ["--", "jj", "diff", "--tool", ""]);
+    insta::assert_snapshot!(output, @r"
+    diffedit3
+    diffedit3-ssh
+    difft
+    kdiff3
+    meld
+    meld-3
+    mergiraf
+    smerge
+    vimdiff
+    vscode
+    vscodium
+    [EOF]
+    ");
+    // Includes :builtin
+    let output = test_env.run_jj_in(dir, ["--", "jj", "diffedit", "--tool", ""]);
+    insta::assert_snapshot!(output, @r"
+    :builtin
+    diffedit3
+    diffedit3-ssh
+    difft
+    kdiff3
+    meld
+    meld-3
+    mergiraf
+    smerge
+    vimdiff
+    vscode
+    vscodium
```

Comment:
I'm not sure whether I want to merge this before or after the release. If we merge it after the release, we could implement something like https://github.com/jj-vcs/jj/issues/4549 to remove non-functional diff editors like `vscode` and `mergiraf` from the list.

Reply:
Thanks for pointing me to it! I think that would help with #4549, yes. I don't think it affects this PR much.